File: frontend/src/enums/biz/customerSaleSettleBizType.ts
import { BaseEnum, BaseEnumItem } from '@/enums/baseEnum';

const CUSTOMER_SALE_SETTLE_BIZ_TYPE: BaseEnum<number, string> = new BaseEnum<number, string>();
CUSTOMER_SALE_SETTLE_BIZ_TYPE.set(
  'OUT_SHEET',
  new BaseEnumItem<number, string>(1, '销售出库单'),
);
CUSTOMER_SALE_SETTLE_BIZ_TYPE.set(
  'SALE_RETURN',
  new BaseEnumItem<number, string>(2, '销售退单'),
);

export { CUSTOMER_SALE_SETTLE_BIZ_TYPE };

File: backend/xingyun-settle/src/main/java/com/lframework/xingyun/settle/enums/CustomerSaleSettleBizType.java
package com.lframework.xingyun.settle.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.lframework.starter.web.core.enums.BaseEnum;

/**
 * 客户销售结算业务类型。
 */
public enum CustomerSaleSettleBizType implements BaseEnum<Integer> {
  OUT_SHEET(1, "销售出库单"), SALE_RETURN(2, "销售退单");

  @EnumValue
  private final Integer code;

  private final String desc;

  CustomerSaleSettleBizType(Integer code, String desc) {

    this.code = code;
    this.desc = desc;
  }

  @Override
  public Integer getCode() {

    return this.code;
  }

  @Override
  public String getDesc() {

    return this.desc;
  }
}
